src/NuGet.Core/NuGet.Configuration/Settings/Items/CertificateItem.cs

```diff
@@ -0,0 +1,115 @@
+                {
+                    return parsedAttribute;
+                }
+
+                return false;
+            }
+        }
+
+        protected override HashSet<string> RequiredAttributes => new HashSet<string>() { ConfigurationConstants.Fingerprint, ConfigurationConstants.HashAlgorithm, ConfigurationConstants.AllowUntrustedRoot };
+
+        public CertificateItem(string fingerprint, HashAlgorithmName hashAlgorithm, bool allowUntrustedRoot = false)
+            : base()
+        {
+            if (string.IsNullOrEmpty(fingerprint))
+            {
+                throw new ArgumentNullException(nameof(fingerprint));
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.Unknown)
+            {
+                throw new ArgumentException(Resources.UnknownHashAlgorithmNotSupported);
+            }
+
+            AddAttribute(ConfigurationConstants.Fingerprint, fingerprint);
+            AddAttribute(ConfigurationConstants.HashAlgorithm, hashAlgorithm.ToString());
+            AddAttribute(ConfigurationConstants.AllowUntrustedRoot, $"{allowUntrustedRoot}");
+        }
+
+        internal CertificateItem(XElement element, SettingsFile origin)
+            : base(element, origin)
+        {
+            if (HashAlgorithm == HashAlgorithmName.Unknown)
+            {
+                throw new NuGetConfigurationException(string.Format(CultureInfo.CurrentCulture, Resources.UserSettings_UnableToParseConfigFile,
+                    string.Format(CultureInfo.CurrentCulture, Resources.UnsupportedHashAlgorithm, Attributes[ConfigurationConstants.HashAlgorithm]),
+                    origin.ConfigFilePath));
+            }
+        }
+
+        internal override SettingBase Clone()
+        {
+            var newItem = new CertificateItem(Fingerprint, HashAlgorithm, AllowUntrustedRoot);
+
+            if (Origin != null)
+            {
+                newItem.SetOrigin(Origin);
+            }
+
+            return newItem;
+        }
+
+        public bool Equals(CertificateItem other)
+        {
+            if (other == null || other.GetType() != GetType())
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return string.Equals(Fingerprint, other.Fingerprint, StringComparison.Ordinal);
+        }
+
+        public bool DeepEquals(CertificateItem other)
+        {
+            if (other == null || other.GetType() != GetType())
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (other.Attributes.Count == Attributes.Count)
+            {
+                return Attributes.OrderedEquals(other.Attributes, data => data.Key, StringComparer.OrdinalIgnoreCase);
```

Comment:
Wouldn't this compare the fingerprint case-insensitively?  How does that jive with [this](https://github.com/NuGet/NuGet.Client/pull/2426/files#diff-cfe7b60de891d179cb60eb67bc62589cR87)?

Reply:
No, this is comparing the key case-insensitively, therefore the word "fingerprint"